packages/typespec-azure-resource-manager/lib/extension/parameters.tsp

```diff
@@ -0,0 +1,223 @@
+namespace Azure.ResourceManager.Extension;
+
+using Http;
+using Rest;
+using Azure.ResourceManager.Private;
+using Azure.ResourceManager.Extension.Private;
+using Azure.ResourceManager.CommonTypes;
+
+/**
+ * The default scope parameter for an extension resource.
+ * @template Type The type of the scope parameter (default is string). This can be used to specify `Azure.Core.armResourceIdentifier` type or other constrained type
+ *
+ * @example
+ * ```typespec
+ * model Employee {
+ *  ...ResourceUriParameter;
+ * }
+ * ```
+ */
+@builtInResource
+model ScopeParameter<Type extends string = string> {
+  @path(#{ allowReserved: true })
+  @key
+  @doc("The fully qualified Azure Resource manager identifier of the resource.")
+  scope: Type;
+}
```

Comment:
Are we going to delete the existing `ExtensionResource`?

Reply:
No